syntax = "proto3";

message PutRequest {
  string key = 1;
  string value = 2;
}

message PutResponse {
}

message GetRequest {
  string key = 1;
}

message GetResponse {
  string value = 1;
}

message GetAllKeysRequest {}

message GetKeyResponse {
  string key = 1;
}

service KeyValueStore {
  rpc put(PutRequest) returns (PutResponse);
  rpc get(GetRequest) returns (GetResponse);
  rpc getAllKeys(GetAllKeysRequest) returns (stream GetKeyResponse);
}